Formula/r/recc.rb

```diff
@@ -0,0 +1,188 @@
+      no_path_rewrite=1
+
+      # Do not retry
+      retry_limit=0
+
+      # Platform properties
+      REMOTE_PLATFORM_OS=<%= platform_os %>
+      REMOTE_PLATFORM_ISA=<%= Hardware::CPU.arch.to_s %>
+    EOS
+  end
+
+  def generate_recc_server(bin_directory)
+    (bin_directory / "recc-server").write <<~EOS
+      #!/usr/bin/env sh
+      #{bin}/buildbox-casd \
+        --local-server-instance=recc-server \
+        --protect-session-blobs \
+        ${1:-#{var}/recc/casd}
+    EOS
+  end
+
+  def generate_recc_wrapper(bin_directory, compiler)
+    (bin_directory / "recc-#{compiler}").write <<~EOS
+      #!/usr/bin/env sh
+      #{bin}/recc $(command -v #{compiler}) $@
+    EOS
+  end
+
+  def buildbox_cmake_args
+    %W[
+      -DCASD=ON
+      -DCASD_BUILD_BENCHMARK=OFF
+      -DCASDOWNLOAD=OFF
+      -DCASUPLOAD=OFF
+      -DFUSE=OFF
+      -DLOGSTREAMRECEIVER=OFF
+      -DLOGSTREAMTAIL=OFF
+      -DOUTPUTSTREAMER=OFF
+      -DRECC=ON
+      -DREXPLORER=OFF
+      -DRUMBA=OFF
+      -DRUN_BUBBLEWRAP=OFF
+      -DRUN_HOSTTOOLS=ON
+      -DRUN_OCI=OFF
+      -DRUN_USERCHROOT=OFF
+      -DTREXE=OFF
+      -DWORKER=OFF
+      -DRECC_CONFIG_PREFIX_DIR=#{etc}/recc
+    ]
+  end
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", *buildbox_cmake_args, *std_cmake_args
+    system "cmake", "--build", "build", "-j", ENV.make_jobs.to_s
+    system "cmake", "--install", "build"
+
+    generate_recc_wrapper bin, "cc"
+    generate_recc_wrapper bin, "c++"
+    generate_recc_wrapper bin, "gcc"
+    generate_recc_wrapper bin, "g++"
+    generate_recc_wrapper bin, "clang"
+    generate_recc_wrapper bin, "clang++"
+
+    generate_recc_server bin
+
+    generate_recc_conf prefix/"etc/recc/"
+  end
+
+  def caveats
+    on_macos do
+      <<~EOS
+        If you are starting the `recc` service via `brew service start`, please ensure the maxfiles limit
+        for `launchctl` is larger than 364. If you are running `recc-server` directly, please ensure
+        `ulimit -Sn` is larger than 364.
+
+         To increase the limit and make the change persist across reboots, you can create a launch daemon
+         using the following steps:
+
+         sudo /usr/libexec/PlistBuddy /Library/LaunchDaemons/limit.maxfiles.plist \\
+          -c "add Label string com.launchd.maxfiles" \\
+          -c "add ProgramArguments array" \\
+          -c "add ProgramArguments: string launchctl" \\
+          -c "add ProgramArguments: string limit" \\
+          -c "add ProgramArguments: string maxfiles" \\
+          -c "add ProgramArguments: string 512" \\
+          -c "add ProgramArguments: string unlimited" \\
+          -c "add RunAtLoad bool true" && \\
+         sudo chown root:wheel /Library/LaunchDaemons/limit.maxfiles.plist && \\
+         sudo chmod 644 /Library/LaunchDaemons/limit.maxfiles.plist && \\
+         sudo launchctl load -w /Library/LaunchDaemons/limit.maxfiles.plist
+      EOS
+    end
+  end
```

Comment:
```suggestion
```
This isn't particular to Homebrew; please remove it.

Reply:
Removed.